### PR TITLE
Backwards compatibility exercise

### PR DESCRIPTION
**Pull request by NoyRobinson**
added option first name and last name for owner

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        fun addOwnerV1(@RequestBody owner: Owner): Response {
+            Response.status(Response.Status.CREATED).build()
+            return Response.ok(owner).build()
+        }
+
+        // create v2
+        @POST
+        @Consumes(MediaType.APPLICATION_JSON)
+        @Path("/newOwner/v2")
+        fun addOwnerV2(@RequestBody owner: Owner): Response {
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]: What's the difference between the 2? They both get the same Owner object. Maybe you can provide some extra implementation about how you use it differently?

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            Response.status(Response.Status.OK).build()
+            return Response.ok(
+                listOf(Owner(id = UUID.randomUUID(), companyId = UUID.randomUUID(), employeeId = UUID.randomUUID(), name = "Noy Robinson", firstName = null, lastName = null, type = "dog", dateOfArrival = Date()))
+            ).build()
+        }
+
+        // retrieve v2
+        @GET
+        @Path("/allOwners/v2")
+        fun getAllOwnersV2(): Response {
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]: So actually the way you implemented it, there is no need for v1/v2. We usually change the version when there are breaking changes. The way you did it, allows for the user to send/receive both ways in the same object. 
So the same version can be used, just need to validate in the code that they are sending 1 or the other.

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            Response.status(Response.Status.ACCEPTED).build()
+            return Response.ok(owner).build()
+        }
+
+        // update v2
+        @PUT
+        @Consumes(MediaType.APPLICATION_JSON)
+        @Path("/updateOwnerType/{ownerId}/v2")
+        fun updateOwnerTypeV2(@PathParam("ownerId") ownerId: Long, @QueryParam("newType") newType: String?): Response {
+            val owner = Owner(id = ownerId, companyId = 12L, employeeId = "123", name = null, firstName = "Noy", lastName = "Robinson", type = "dog", dateOfArrival = Date())
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt:49]: Hey, so I think maybe I wasn't clear on the differences of implementation. If you want to use the v1/v2 versioning which creates a new API - then using the same object isn't necessary. 
Expanding the original Pet to hold both name and firstName, lastName then you don't need the v1/v2.

Comment (NoyRobinson) [reply src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt:49]: I expanded it to hold both, so I don't need to have 2 versions of getting an update with first name and last name and getting an update with full name?

**Review by brainpicker — APPROVED** [review]

**Pull request closed without merging**